kpi-tracker/src/main.rs

```diff
@@ -362,21 +425,38 @@ async fn process_block(
         height:     block_info.block_height,
     };
 
-    let mut accounts: BTreeMap<AccountAddress, AccountDetails> = BTreeMap::new();
+    let mut accounts: BTreeMap<CanonicalAccountAddress, AccountDetails> = BTreeMap::new();
     let mut account_transactions: BTreeMap<TransactionHash, TransactionDetails> = BTreeMap::new();
     let mut contract_modules: BTreeMap<ModuleRef, ContractModuleDetails> = BTreeMap::new();
     let mut contract_instances: BTreeMap<ContractAddress, ContractInstanceDetails> =
         BTreeMap::new();
+    let mut transaction_account_relations: BTreeSet<TransactionAccountRelation> = BTreeSet::new();
+    let mut transaction_contract_relations: BTreeSet<TransactionContractRelation> = BTreeSet::new();
 
     let block_events = get_block_events(node, block_info.block_hash).await?;
     block_events
         .try_for_each(|be| {
             match be {
                 BlockEvent::AccountCreation(address, details) => {
-                    accounts.insert(address, details);
+                    accounts.insert(CanonicalAccountAddress::from(address), details);
                 }
-                BlockEvent::AccountTransaction(hash, details) => {
+                BlockEvent::AccountTransaction(
+                    hash,
+                    details,
+                    affected_accounts,
+                    affected_contracts,
+                ) => {
                     account_transactions.insert(hash, details);
+
+                    if !affected_accounts.is_empty() {
+                        transaction_account_relations
+                            .append(&mut BTreeSet::from_iter(affected_accounts.into_iter()));
```

Comment:
```suggestion
                            .extend(affected_accounts.into_iter());
```

See https://doc.rust-lang.org/stable/std/iter/trait.Extend.html